### PR TITLE
`scene_texture`s

### PR DESCRIPTION
**Pull request by AngheloAlf**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

As commented in Discord, there are a bunch of scenes/rooms that references the segment 6 for loading textures.

After investigating it, I realised those are [references to the scene_texture files](https://github.com/zeldaret/mm/blob/44ccdbe0d5bc2e3aad872613070106dae1de7a4e/src/code/z_scene.c#L347), so after a bit of trial and error, I made XMLs for those files and added references to the corresponding scenes (as a ZAPD's ExternalFile)

I also uncommented the global ExternalFile reference in the ZAPD configuration file to gameplay_keep, so every extracted asset making a reference to segment 4 should use the corresponding symbol instead of a raw segmented pointer.

Finally, I left the script I used to create the scene_texture XMLs in the tools folder as a reference for anybody interested.

### CODE REVIEW

File: assets/xml/misc/scene_texture_03.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<Root>
+    <File Name="scene_texture_03" Segment="6">
+        <Blob Name="scene_texture_03_Blob_002D40" Size="0x4800" Offset="0x0"/>
```
Comment (EllipticEllipsis) [inline assets/xml/misc/scene_texture_03.xml:3]: Most of this seems to be trees in various formats, but beyond the first couple it's not really guessable, annoyingly.

**Review by EllipticEllipsis — COMMENTED** [review]
Nice job, although dealing with these is probably going to be a pain since a lot of them use external palettes... Could you add a note to each stating where it is used? I assume it wouldn't be too much trouble for the script to do that.

File: tools/scene_texture_xml_gen.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (AngheloAlf) [reply tools/scene_texture_xml_gen.py:1]: I left it as a reference, but I can remove it if desided

File: assets/xml/misc/scene_texture_01.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        <Texture Name="scene_texture_01_Tex_001900" OutName="tex_001900" Format="ci8" Width="16" Height="16" Offset="0x1900"/>
+        <Texture Name="scene_texture_01_Tex_001A00" OutName="tex_001A00" Format="ci8" Width="32" Height="32" Offset="0x1A00"/>
+        <Texture Name="scene_texture_01_Tex_001E00" OutName="tex_001E00" Format="ci8" Width="32" Height="32" Offset="0x1E00"/>
+        <Texture Name="scene_texture_01_Tex_002200" OutName="tex_002200" Format="ci8" Width="16" Height="16" Offset="0x2200"/>
+        <Texture Name="scene_texture_01_Tex_002300" OutName="tex_002300" Format="ci8" Width="16" Height="32" Offset="0x2300"/>
+        <Texture Name="scene_texture_01_Tex_002500" OutName="tex_002500" Format="ci8" Width="16" Height="64" Offset="0x2500"/>
+        <Texture Name="scene_texture_01_Tex_002900" OutName="tex_002900" Format="ci8" Width="16" Height="64" Offset="0x2900"/>
+        <Texture Name="scene_texture_01_Tex_002D00" OutName="tex_002D00" Format="i8" Width="8" Height="8" Offset="0x2D00"/>
+        <!-- <Blob Name="scene_texture_01_Blob_002D40" Size="0x100" Offset="0x2D40"/> -->
```
[the remark this answers is not in the document]
Comment (AngheloAlf) [reply assets/xml/misc/scene_texture_01.xml:19]: Yeah, I agree it is frustrating, but there isn't anything we could do about it :s

File: tools/scene_texture_xml_gen.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hensldm) [reply tools/scene_texture_xml_gen.py:1]: My thoughts would be to just not add it to the repo unless it is needed by others. It helps prevent clutter, and don't have to worry about keeping it up to date and documented in tools.md.

**Review by hensldm — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

File: tools/scene_texture_xml_gen.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kenix3) [reply tools/scene_texture_xml_gen.py]: I say keep it in the repo, but under a new reference subdirectory.

**Review by Kenix3 — APPROVED** [review]
I think we should add the script back in a new subirectory, but no reason this PR can't be merged.

**Pull request merged**